### PR TITLE
Use statusMessage returned by endpoint

### PR DESCRIPTION
**Pull request by sleewoo**
for cases when endpoints returns a descriptive error message

### CODE REVIEW

**Comment by sindresorhus** [comment]
Can you include a test?

File: index.js [hunk 1]
```diff
@@ -627,8 +627,12 @@ got.ParseError = class extends StdError {
 };
 
 got.HTTPError = class extends StdError {
-	constructor(statusCode, headers, opts) {
-		const statusMessage = http.STATUS_CODES[statusCode];
+	constructor(statusCode, statusMessage, headers, opts) {
+		if (statusMessage) {
+			statusMessage = statusMessage.replace(/\n|\r/gm, ' ');
```
Comment (sindresorhus) [inline index.js]: `/\n|\r/gm` => `/\r?\n/g`

Comment (sleewoo) [reply index.js]: yep, much better

**Comment by sleewoo** [comment]
cranking a test

**Comment by lukechilds** [comment]
I support this change but just wanna point out potential breakage.

Any existing user code like:

```js
if (err.statusMessage === 'Too Many Requests') {
  handleAPILimit();
}
```

will fail after this change if the API returns it's own error message.

**Comment by sindresorhus** [comment]
@lukechilds I'll make sure to point that out in the release notes.

**Comment by sindresorhus** [comment]
Thanks @sleewoo :)

**Pull request merged**